Request code review feature (#1028)
Hacker0x01/docs.hackerone.com: docs/organizations/request-code-review.md

---
title: "Request Code Review (BETA)"
path: "/organizations/request-code-review.html"
id: "organizations/request-code-review"
---

> In addition to confirming a vulnerability is has been fixed by requesting a [Retest](retest.html), you can also request a code review to verify any source code updates adequately address the reported vulnerability. These are conducted by the HackerOne Code Review Community: a group of rigorously-vetted, specialized software engineers and security experts HackerOne created as part of the [acquisition of PullRequest.com](https://www.hackerone.com/press-release/hackerone-acquires-pullrequest-power-developer-first-security-testing-solutions). You can learn more about this specialized cohort of the HackerOne Community [here](https://www.pullrequest.com/reviewers/).

Author: ```suggestion
In addition to confirming a vulnerability has been fixed by requesting a [Retest](retest.html), you can also request a code review to verify any source code updates adequately address the reported vulnerability. These are conducted by the HackerOne Code Review Community: a group of rigorously-vetted, specialized software engineers and security experts HackerOne created as part of the [acquisition of PullRequest.com](https://www.hackerone.com/press-release/hackerone-acquires-pullrequest-power-developer-first-security-testing-solutions). You can learn more about this specialized cohort of the HackerOne Community [here](https://www.pullrequest.com/reviewers/).
```

> In addition to confirming a vulenerability is has been fixed by requesting a [Retest](retest.html), you can also request a code review to verify any source code updates adequately address the reported vulneratiblity. These are conducted by the HackerOne Code Review Community: a group of rigorously-vetted, specialized software engineers and security experts HackerOne created as part of the [acquisition of PullRequest.com](https://www.hackerone.com/press-release/hackerone-acquires-pullrequest-power-developer-first-security-testing-solutions). You can learn more about this specialized cohort of the HackerOne Community [here](https://www.pullrequest.com/reviewers/).

Author: ```suggestion
In addition to confirming a vulnerability is has been fixed by requesting a [Retest](retest.html), you can also request a code review to verify any source code updates adequately address the reported vulnerability. These are conducted by the HackerOne Code Review Community: a group of rigorously-vetted, specialized software engineers and security experts HackerOne created as part of the [acquisition of PullRequest.com](https://www.hackerone.com/press-release/hackerone-acquires-pullrequest-power-developer-first-security-testing-solutions). You can learn more about this specialized cohort of the HackerOne Community [here](https://www.pullrequest.com/reviewers/).
```

Author: I swear I can spell.

The **Request code review** feature allows you to send a limited code snippet patch to be reviewed as generated by `git diff`.

> **Note:** This feature is currently in open beta.

### How it Works  

> To have request a patch code review:

Author: ```suggestion
To request a patch code review:
```

### FAQs
Question | Answer
-------- | -------
Can the person reviewing the code snippet see the contents of the report? | No. Further, identities of the reporter(s) and members of your program team are also NOT disclosed to the engineer reviewing the code.
How does HackerOne vet the reviewers? | All members of the code review community are background-checked, interviewed for technical proficiency & professionalism, sign confidentiality & personal inventions assignment agreements, and are based the U.S., U.K., or Canada.
How much of the source code is exposed to reviewers? | Only what is contained within the diff snippet. These contain whatever lines of code have been changed and, where applicable, a few lines of code above and below the changed lines to provide additional context for the impacted lines of code.
If a reviewer finds a problem with the proposed fix, can I submit another diff snippet with updated code? | Yes. We encourage it!
Can I respond to the reviewer or ask questions? | Yes. Simply respond with an [Internal](/report-components.html#timeline) comment.

> What programming languages, frameworks, libraries and platforms do you support? | All types of code are supported. This includes Infrastructure as Code (IaS) and Configuration as Code (CaC). It does help our community if you expand the Context to include version(s) of a programming language or framework if relevant to determine correctness. Our systems determine the type of code submitted by the file types included as part of the diff snippet and assign to experts in our community accordingly.

Author: ```suggestion
What programming languages, frameworks, libraries and platforms do you support? | All types of code are supported. This includes Infrastructure as Code (IaS) and Configuration as Code (CaC). It does help our community if you expand the Context to include version(s) of a programming language or framework if relevant to determine correctness. Our systems detect the type of code submitted by the file types included as part of the diff snippet and assign to experts in our community accordingly.
```
